File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        print(matrix)
        for i in range(len(matrix)):
            for j in range(i):
                matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]
                print(i,j,matrix)
        print(matrix)
        for i in matrix:
            i[:]=i[::-1]


        

File: 0048-rotate-image/0048-rotate-image.c
void display(int** matrix, int matrixSize, int* matrixColSize) {
    for(int i=0;i<matrixSize;i++){
        for(int j=0;j<*matrixColSize;j++){
            printf("%d ",matrix[i][j]);
        }
        printf("\n");
    }
    printf("\n");
}
void rotate(int** matrix, int matrixSize, int* matrixColSize) {
    display(matrix,matrixSize,matrixColSize);
    for(int i=0;i<matrixSize;i++){
        for(int j=i+1;j<*matrixColSize;j++){
            //printf("%d %d\n",i,j);
            int temp=matrix[i][j];
            matrix[i][j]=matrix[j][i];
            matrix[j][i]=temp;
        }
        //printf("\n");
    }
    //printf("\n");
    for(int i=0;i<matrixSize;i++){
        for(int j=0;j<*matrixColSize/2;j++){
            int temp=matrix[i][j];
            matrix[i][j]=matrix[i][*matrixColSize-1-j];
            matrix[i][*matrixColSize-1-j]=temp;
        }
    }
    display(matrix,matrixSize,matrixColSize);
    
}